executor/show_test.go

```diff
@@ -175,6 +175,31 @@ func (s *testSuite) TestShow(c *C) {
 			"  `id` int(11) DEFAULT NULL\n"+
 			") ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin AUTO_INCREMENT=4",
 	))
+	// for issue https://github.com/pingcap/tidb/issues/4678
+	tk.MustExec("insert into show_auto_increment values(10)")
+	tk.MustQuery(`show create table show_auto_increment`).Check(testutil.RowsWithSep("|",
```

Comment:
If the table doesn't have a column with auto_increment attribute, MySQL will ignore the table option.
```sql
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 89
Server version: 5.7.20-0ubuntu0.16.04.1 (Ubuntu)

Copyright (c) 2000, 2017, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> create table show_auto_increment (id int) auto_increment=4;
Query OK, 0 rows affected (0.02 sec)

mysql> show create table show_auto_increment;
+---------------------+---------------------------------------------------------------------------------------------------------+
| Table               | Create Table                                                                                            |
+---------------------+---------------------------------------------------------------------------------------------------------+
| show_auto_increment | CREATE TABLE `show_auto_increment` (
  `id` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
+---------------------+---------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> 
```

Reply:
Yep, but it's the old logical in TiDB. It handles in PR 3747.
I will fix it.